addons/account/models/ir_attachment.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    def unlink(self):
+        # Prevent unlink journal entrie attachment
+        move_ids = self.filtered(lambda x: x.res_model == 'account.move' and x.res_id).mapped('res_id')
+        if move_ids:
+            move_count = self.env['account.move'].sudo().search_count([('id', 'in', move_ids),
```

Comment:
Why `sudo()`? To delete an attachment of an `account.move` the user must have _read access_ to that move, right? :thinking: